components/LoginForm.vue

```diff
@@ -38,7 +86,81 @@ const user = reactive<User>({
   password: '',
 })
 
-const isDisabled = computed(() => user.username !== '' && user.password !== '')
+const checkIsValidUsername = computed(() => {
+  let isUsernameValid = false
+
+  isUsernameValid = /^[A-Za-z0-9]*$/.test(user.username)
+
+  return isUsernameValid
+})
+
+const checkIsValidPassword = computed(() => {
```

Comment:
const checkIsValidPassword = computed(() => hasMinimumLength.value &&
    hasNumber.value &&
    hasUppercase.value &&
    hasLowercase.value &&
    hasSpecial.value)
    
    if they are booleans you no longer need to check if they are